hoomd/md/methods.py

```diff
@@ -580,6 +583,144 @@ def _attach(self):
+
+        # set defaults
+        self._param_dict.update(param_dict)
+
+    def _attach(self):
+        if not self._manifold._added:
+            self._manifold._add(self._simulation)
+        else:
+            if self._simulation != self._manifold._simulation:
+                raise RuntimeError("{} object's manifold is used in a "
+                                   "different simulation.".format(type(self)))
+        if not self.manifold._attached:
+            self.manifold._attach()
+
+        # initialize the reflected c++ class
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            if self.manifold.name == "Cylinder":
+                my_class = _md.TwoStepRATTLENVECylinder
+            elif self.manifold.name == "Diamond":
+                my_class = _md.TwoStepRATTLENVEDiamond
+            elif self.manifold.name == "Ellipsoid":
+                my_class = _md.TwoStepRATTLENVEEllipsoid
+            elif self.manifold.name == "Gyroid":
+                my_class = _md.TwoStepRATTLENVEGyroid
+            elif self.manifold.name == "Plane":
+                my_class = _md.TwoStepRATTLENVEPlane
+            elif self.manifold.name == "Primitive":
+                my_class = _md.TwoStepRATTLENVEPrimitive
+            elif self.manifold.name == "Sphere":
+                my_class = _md.TwoStepRATTLENVESphere
+            else:
+                raise AttributeError("Manifold class not implemented")
+        else:
+            if self.manifold.name == "Cylinder":
+                my_class = _md.TwoStepRATTLENVECylinderGPU
+            elif self.manifold.name == "Diamond":
+                my_class = _md.TwoStepRATTLENVEDiamondGPU
+            elif self.manifold.name == "Ellipsoid":
+                my_class = _md.TwoStepRATTLENVEEllipsoidGPU
+            elif self.manifold.name == "Gyroid":
+                my_class = _md.TwoStepRATTLENVEGyroidGPU
+            elif self.manifold.name == "Plane":
+                my_class = _md.TwoStepRATTLENVEPlaneGPU
+            elif self.manifold.name == "Primitive":
+                my_class = _md.TwoStepRATTLENVEPrimitiveGPU
+            elif self.manifold.name == "Sphere":
+                my_class = _md.TwoStepRATTLENVESphereGPU
+            else:
+                raise AttributeError("Manifold class not implemented")
+
+        self._cpp_obj = my_class(self._simulation.state._cpp_sys_def,
+                                 self._simulation.state._get_group(self.filter), 
+                                 self.manifold._cpp_manifold, False, self.eta)
+
+        # Attach param_dict and typeparam_dict
+        super()._attach()
+
+    @property
+    def manifold(self):
+        return self._manifold
+
+    @manifold.setter
+    def manifold(self, value):
+        if self._attached:
+            raise AttributeError("manifold cannot be set after scheduling.")
+        else:
+            self._manifold = validate_manifold(value)
+
+    @property
+    def _children(self):
```

Comment:
Are children just the manifolds? If so, maybe rename accordingly? If not, this needs a short docstring (even for an internal property) indicating what the children are.